File: src/de/esymetric/jerusalem/ownDataRepresentation/fileSystem/NodeIndexFile.java
package de.esymetric.jerusalem.ownDataRepresentation.fileSystem;

import java.util.List;

public interface NodeIndexFile {
	public class NodeIndexNodeDescriptor {
		public int id;
		public int latInt;
		public int lngInt;
	}
	
	public int setID(double lat, double lng, int id);
	public int getID(double lat, double lng);
	public List<NodeIndexNodeDescriptor> getIDPlusSourroundingIDs(double lat, double lng, int radius);
	public int getCapacity();
	public void close();
	public float getWriteCacheHitRatio();
	public float getReadCacheHitRatio();
	public int getCacheSize();
	public int getMaxCacheSize();
}

File: src/main/java/de/esymetric/jerusalem/ownDataRepresentation/fileSystem/NodeIndexFile.kt
package de.esymetric.jerusalem.ownDataRepresentation.fileSystem

interface NodeIndexFile {
    class NodeIndexNodeDescriptor {
        var id = 0
        var latInt = 0
        var lngInt = 0
    }

    fun setID(lat: Double, lng: Double, id: Int): Int
    fun getID(lat: Double, lng: Double): Int
    fun getIDPlusSourroundingIDs(lat: Double, lng: Double, radius: Int): List<NodeIndexNodeDescriptor>
    val capacity: Int
    fun close()
    val writeCacheHitRatio: Float
    val readCacheHitRatio: Float
    val maxCacheSize: Int
}